### DISCUSSION TITLE
Do we use TCP back pressure by default?

### DISCUSSION THREAD

**Discussion by kitkars** [pgjdbc/r2dbc-postgresql#654] [General]
The default fetch size is 0. So Does R2DBC rely on TCP back pressure?
If not, how does it work?

**Comment by mp911de** [comment]
Yes, we use TCP back pressure (even if fetch size is set). This works by not reading from the TCP channel unless the driver sees more demand. That leaves TCP packets in recv buffers. As soon as these are full, the server saturates TCP send buffers. Then, the server needs to react to full send buffers by e.g. not sending more data.

**Reply by kitkars** [reply]
Hi Mark.. 1 follow up question if you do not mind. When the receiver & sender buffer are full, does postgres db server know how to react? For ex: If a client does a "select * from users" with 100 million records with r2dbc and too slow to consume, the receiver buffer will become full with, say, 1000 records. In this case, postgres will not send any more data? Not sure what kind of negative impact a db server can have in this case! in terms of db resources etc.

**Reply by mp911de** [reply]
I don't know, that's an excellent question for https://www.postgresql.org/list/pgsql-general/
